packages/ckeditor5-mention/src/mentionui.js

```diff
@@ -161,7 +163,7 @@ export default class MentionUI extends Plugin {
 			}
 
 			const minimumCharacters = mentionDescription.minimumCharacters || 0;
-			const feedCallback = typeof feed == 'function' ? feed.bind( this.editor ) : createFeedCallback( feed );
+			const feedCallback = typeof feed == 'function' ? feed.bind( this.editor ) : createFeedCallback( feed, dropdownLimit );
```

Comment:
The way you applied the fix will work only for one out of two use cases:

*   ✅ Passing a simple array of mentions like you did in unit tests.
*   ❌ Passing a custom callback which in turn return an array of mentions.

See [https://ckeditor.com/docs/ckeditor5/latest/api/module_mention_mention-MentionFeed.html](https://ckeditor.com/docs/ckeditor5/latest/api/module_mention_mention-MentionFeed.html) to see some examples.

Now I see that you applied fix to the data of the feed - this is incorrect. You want to limit the entries in _view only_. The view handler is common for any methods mentions would be fetch.

I pushed a dirty unit test that will show the issue in the `ck/10479-mention-dropdownlimit-failing-test` branch.